gtsam/geometry/tests/testRot3.cpp

```diff
@@ -663,14 +664,12 @@ TEST(Rot3, ClosestTo) {
 
 /* ************************************************************************* */
 TEST(Rot3, angle) {
-  Rot3 R1 = Rot3::Ypr(0, 0, 0);
-  
-  Rot3 R2 = Rot3::Ypr(M_PI/4, 0, 0);
+  Rot3 R1 = Rot3::Ypr(M_PI/4, 0, 0);
 
-  Unit3 expectedAxis(0, 0, 1);
-  double expectedAngle = M_PI/4;
+  Unit3 expectedAxis(-0.350067, -0.472463, 0.808846);
```

Comment:
Again, not a great test. How about just adding three tests for each of yaw, pitch, roll? It’ll be interesting in itself as, thinking about it, there is a sign ambiguity that we probably need to resolve. For example, guarantee the angle is in `(-pi, pi]`

Reply:
I'm confused. Am I still supposed to use the rotation defined at the top of the file?

If I understand correctly, a good set of tests would be those that check for roll, pitch and yaw individually using numbers that make it easy to verify manually that what is happening is indeed correct. Isn't this the same as my original test?
After that, using the rotation defined at the top would involve another, non-trivial test(?). I am not sure where the sign ambiguity fits in here. Wouldn't that need a specific rotation matrix that is obtuse and which would require `ToAxisAngle` to correctly return an acute angle?

Reply:
more tests are better, so do the three yaw/pitch/roll ones.
The one at the top , yes, but structure the test that the answer is obvious, rather than copy from output:
```
Unit3 axis; double angle;
std::tie(axis,angle) = R.axisAngle(0);
EXPECT(assert_equal(Unit3(0.1, 0.4, 0.2)));
EXPECT_DOUBLES_EQUAL(Vector3(0.1, 0.4, 0.2).norm());
```
Sign ambiguity: you can construct a test case with AxisAngle that has an angle `theta` larger than pi, and axisAngle should then return `theta-2*pi`. And vice-versa, angle < -pi.